setupKAppNavTestEnv.sh

```diff
@@ -0,0 +1,172 @@
+    echo "   <platform> specifies one of: okd, ocp, minishift or minikube. Default is okd"
+    echo "   <dockerUsername> specifies your docker username"
+    echo
+    exit 0
+fi
+    
+# get positional parms
+platformURL=$1
+platformUsername=$2
+platformPassword=$3
+platform=$4
+dockerID=$5
+
+if [ x$platform = 'xminikube' ] ; then
+    kubectl=$(kubectl)
+    if [ $? -ne 0 ]; then
+        echo "Error: kubectl not found. Make sure minikube is running."
+        echo ""
+        exit 1
+    fi
+else 
+    # make sure to logout first, in case it already login to different platform
+    oc logout
+    # login to target platform
+    echo "oc login -u $platformUsername -p $platformPassword $platformURL"
+    oc login -u $platformUsername -p $platformPassword $platformURL
+    if [ $? -eq 0 ]; then
+        echo "########## OC login successfully. ##########"
+    else
+        echo "########## OC login failed, exiting. ##########"
+        exit 1
+    fi
+fi
+
+# At least check if previous KAppNav already install on user namespace
+# If there is then uninstall first as having 2 instances install on same platform won't work well
+
+# check if KAppNav has been installed using docker id namespace
+./test/isDeployed.sh $dockerID
+if [ $? -eq 0 ]; then
+    echo "KAppNav already installed on namespace $dockerID"
+    echo "Uninstalling KAppNav from $dockerID namespace"
+    kubectl delete -f ../operator/kappnav-delete-CR.yaml -n $dockerID --now
+    kubectl delete -f ../operator/kappnav-delete.yaml -n $dockerID
+    kubectl delete namespace $dockerID
+    if [ $? -eq 0 ]; then
+        echo "########## Uninstall successfully. ##########"
+    else 
+        echo "########## Uninstall failed, exiting ##########"
+        exit 1
+    fi
+fi
+
+# check if KAppNav has been installed on kappnav namespace
+# because if either AppNav or KAppNav already install on the platform you are using then thing won't work well
+./test/isDeployed.sh kappnav
+if [ $? -eq 0 ]; then
+    echo "KAppNav already installed on namespace kappnav"
+    echo "Uninstalling KAppNav from kappnav namespace"
+    ./uninstall.sh
+    if [ $? -eq 0 ]; then
+        echo "########## Uninstall successfully. ##########"
+    else 
+        echo "########## Uninstall failed, exiting ##########"
+        exit 1
+    fi
+fi
+
+# build kappnav
+echo "########## Building KAppNav ########## "
+./build.sh
+if [ $? -eq 0 ]; then
+    echo "########## Build successfully. ##########"
+else
+    echo "########## Build failed, exiting. ##########"
+    exit 1
+fi
+
+# push all images to docker hub
+echo "########## Pushing all KAppNav images to docker hub of $dockerID ########## "
+./pushimages.sh $dockerID
+if [ $? -eq 0 ]; then
+    echo "########## Pushed KAppNav images successfully. ##########"
+else
+    echo "########## Pushed KAppNav images failed, exiting. ##########"
+    exit 1
+fi
+
+# installing to target platform
+./install.sh $dockerID $platform
```

Comment:
This should be "kappnav", not $dockerID

Reply:
Roger, I make it docker id so user can use it to publish it to their own docker hub, but it will still be install on kappnav namespace. This will allow user to test their kappnav dev code, not just using the kappnav docker images only. If user want to user publish kappnav image then they can just use kappnav as docker id. What do you think?

Reply:
Sorry, you're right.  I mistook it for the target namespace.